### PR TITLE
SAMSON-254 added deployUrl as a fourth param to Jenkins builds

### PR DESCRIPTION
**Pull request by zehua**
- added a new build param `deployUrl` for Jenkins builds
- added JENKINS related env vars into .env.example

/cc @zendesk/samson @zendesk/vulcan

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: https://zendesk.atlassian.net/browse/SAMSON-254

### Risks
- Level: Low

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -106,3 +106,8 @@ GITHUB_TOKEN=
 ## Kubernets sidecar
 # SECRET_SIDECAR_IMAGE= # optional, path to the docker image that built from the secret_puller service.  service will be disabled w/o it
 
+
+## Jenkins, required for triggering Jenkins build after deployment
```
Comment (grosser) [inline .env.example]: we try to use the `required` work for things that need to be filled out, how about `Jenkins, optional for triggering Jenkins build after deployment`

File: plugins/jenkins/app/models/samson/jenkins.rb [hunk 2]
```diff
@@ -1,4 +1,3 @@
-# rubocop:disable Metrics/LineLength
```
Comment (grosser) [inline plugins/jenkins/app/models/samson/jenkins.rb]: ❤️

File: plugins/jenkins/app/models/samson/jenkins.rb [hunk 3]
```diff
@@ -17,7 +16,12 @@ def initialize(job_name, deploy)
     def build
       opts = {'build_start_timeout' => 60}
       originated_from = deploy.project.name + '_' + deploy.stage.name + '_' + deploy.reference
-      client.job.build(job_name, {'buildStartedBy' => deploy.user.name, 'originatedFrom' => originated_from, 'commit' => deploy.job.commit}, opts).to_i
+      client.job.build(job_name, {
+        buildStartedBy: deploy.user.name,
+        originatedFrom: originated_from,
+        commit: deploy.job.commit,
+        deployUrl: AppRoutes.url_helpers.project_deploy_url(deploy.project, deploy)
```
Comment (grosser) [inline plugins/jenkins/app/models/samson/jenkins.rb]: `deploy.url`

File: plugins/jenkins/test/models/samson/jenkins_test.rb [hunk 4]
```diff
@@ -19,7 +19,7 @@ def stub_job_detail
   def stub_build_with_parameters
     stub_request(:post, "http://www.test-url.com/job/test_job/buildWithParameters").
       with(
-        body: {"buildStartedBy" => "Super Admin", "originatedFrom" => "Project_Staging_staging", "commit" => "staging"},
+        body: {"buildStartedBy" => "Super Admin", "originatedFrom" => "Project_Staging_staging", "commit" => "staging", "deployUrl" => "http://www.test-url.com/projects/foo/deploys/178003093"},
```
Comment (grosser) [inline plugins/jenkins/test/models/samson/jenkins_test.rb]: might be better to use `#{deploy.id}` or `#{Deploy.first.id}` or something like that here ... that id might change randomly

**Comment by grosser** [comment]
looks good!

File: .env.example (hunk 1 above)
Comment (zehua) [inline .env.example]: updated

File: plugins/jenkins/app/models/samson/jenkins.rb (hunk 3 above)
Comment (zehua) [inline plugins/jenkins/app/models/samson/jenkins.rb]: done

File: plugins/jenkins/test/models/samson/jenkins_test.rb (hunk 4 above)
Comment (zehua) [inline plugins/jenkins/test/models/samson/jenkins_test.rb]: done

**Comment by hyperair** [comment]
:+1:

**Pull request merged**